### DISCUSSION TITLE
A methodology to categorize exam series.

### DISCUSSION THREAD

**Discussion by Chevenger1** [Emory-HITI/Niffler#388] [Q&A] [answered]
I'm working on a project about text classification and autocorrect. Do you think this experience is relevant to our current project? (I have also worked on other projects.) The problem we are currently facing is similar to text classification, so I believe my skills will be useful.

These are few doubts I have :
1. Are there any conventions for series naming? 
2. Where can I get a standard series name (standardized predefined protocol) for all the protocols?
3. Where can I get a list of parameters that should be present so images qualify as DICOM standards? 
4. Can a series belong to multiple protocols, e.g., brain protocol or trauma protocol? 
5. Do I have to create my own dataset for a classification problem of series name to protocol?"

**Comment by pradeeban** [comment, marked as answer]
Although it does not sound like directly related, please include that in your proposal.

Answers to your questions:


    Are there any conventions for series naming?
        No. Though there is a weak association between series name and series classification (i.e. orientation and contrast-weighting), they can be unreliable, generally speaking, since they are free-text fields. 
    Where can I get a standard series name (standardized predefined protocol) for all the protocols?
        We will provide the "pre-defined protocol" rules from which the student will categorize/classify given MR series. "Brain Routine" and "Brain Tumor" are two examples of protocols, and may contain ~10 series each. A simple example of a series category is [Orientation, Image-Weighting] = [Transverse, T1]. The series Dicom metadata or image pixel data can be used to accurately assign an unknown series to this category. 
    Where can I get a list of parameters that should be present so images qualify as DICOM standards?
        Any basic MRI text (print or online) discussing "MR image contrast" will provide parameter ranges associated with various image-weightings, such as T1, T2, T2-FLAIR, contrast-enhanced T1, diffusion-weighted, etc. Many parameters are contained in the DICOM header, but entries may vary due to MR manufacturer. For this project, we will limit the categorization to brain sequences.
    Can a series belong to multiple protocols, e.g., brain protocol or trauma protocol?
        Yes. Most of the brain protocols have common series. For this project, we will provide the protocol template to use for a given exam
    Do I have to create my own dataset for a classification problem of series name to protocol?"
        We will provide the exam datasets (either public or anonymized in-house data). The goal would be to create an accurate and reproducible model to classify the various series of brain exams. Naturally, the model can be trained and validated on known (labeled) datasets, then tested on blinded exam datasets.

**Reply by Chevenger1** [reply]
Could you please provide any available datasets and rules? This would enable me to create pseudo-code and plan my approach more accurately. Thank you.

**Reply by pradeeban** [reply]
Can you check for datasets from https://www.cancerimagingarchive.net/? Emory data cannot be shared without additional requirements. But with the accepted contributors, we can share more data.

**Reply by Chevenger1** [reply]
Can you check if I have understood the project goals correctly? 
Firstly, the Niffler module will identify the series in a brain sequence using a deep learning approach on DICOM metadata or pixel data. It will then find the orientation and weighting of each series. Based on the identified series and given predefined protocols, we will categorize the MRI study (not series) into different brain protocols. We will check if the DICOM parameters meet certain threshold values for acceptance. Finally, we will determine whether any series in the MRI study are repeated or missing altogether.

**Reply by Chevenger1** [reply]
Are we going to use deep learning only for finding the series categories of [Orientation, Image-Weighting]? After that, will we be using protocol rules for classification?
Can you please share an example of  a protocol rule?

**Reply by pradeeban** [reply]
> we will categorize the MRI study (not series) into different brain protocols. 

A study is composed of several series (and a series is composed of several image instances). So categorizing a study and a series in it are essentially the same.

Other than that observation, your interpretation is correct.

**Reply by Chevenger1** [reply]
But 
If a series can belong to multiple protocols, would classifying it result in a multi-label classification problem and no definitive answer? Conversely, would categorizing a study with multiple series give us only one protocol with a confidence score above the threshold?

Is it necessary to utilize predefined protocol rules if we adopt the series categorization method described above? As a neural network, it easily categorizes a study into any brain protocol by inputting sequence-related information such as series sequence (T1, T2, FLAIR, etc.), plane (axial, sagittal, etc.), slice, mode, gap, etc.( multi-class classification problem).

**Reply by pradeeban** [reply]
Good questions. I understand your points/concerns on a theoretical basis. But to know how it will be in practice (whether only one protocol with confidence score above the threshold or it will result in a multi-label classification problem), I have forwarded your questions to Puneet. I will reply again when he responds with a definitive answer.

Please include this discussion and your other contributions in your GSoC proposal.

**Reply by pradeeban** [reply]
As a first pass, a pre-defined protocol will be provided, listing the required series/sequences and expected parameter thresholds for an exam. The input is an arbitrary clinical MRI exam (which consists of a set of unknown/unlabeled series). We would like to label/categorize each unknown series as "belonging to", "not belonging to", or "repeated" (or "indeterminate") in regard to the pre-defined protocol. We'd also like to identify any "missing" series. Once categorized, compliance with expected parameter thresholds should be determined (yes or no). One can potentially compute a final "series compliance score" by assigning 1 point each to successful categorization and compliance with parameter expectations. 

Examples of parameter expectations are: slice thickness, gap, voxel-size, etc

**Reply by Chevenger1** [reply]
I emailed the proposal to my prospective mentors, and CC'd you. I hope to receive approval from my mentors. I will make any corrections that are needed within the deadline. I apologize for the inconvenience caused by my delay in sending it.

**Reply by pradeeban** [reply]
@Chevenger1 I went through your proposal. I liked it! 

Please note the mentors (specifically, Puneet and Tony) will start reviewing the proposals and get back to you with their comments once the application period is over. (So, if possible, be available for clarifications/amendments).

**Comment by Chevenger1** [comment]
Thank you for providing comprehensive clarification regarding my doubts throughout the application period. Based on your suggestions, I have readjusted my tasks for the coding period and created a new timeline. Please feel free to contact me via email, WhatsApp, or any preferred medium of communication for clarifications or amendments to my proposal. I am excited to contribute to Emory BMI this summer and do my part for open source.

**Reply by pradeeban** [reply]
Awesome! Thanks for the updates.
